NestJS/src/Templates/template.listener.ts

```diff
@@ -2,20 +2,50 @@ import { Controller } from '@nestjs/common';
 import * as Amqp from "amqp-ts";
 import { TemplateService } from './template.service';
 
-var connection = new Amqp.Connection("amqp://localhost");
-var exchange = connection.declareExchange("test3", 'direct', { durable: false });
-var queue = connection.declareQueue('');
+const exchange = connection.declareExchange("exchangeForTemplate", 'direct', { durable: false });
+const queue = connection.declareQueue('template');
 
 @Controller('template')
 export class TemplateListener {
     constructor(private readonly templateService: TemplateService){
+      this.listenQueue();
+    }
+
+    public distributionTasks(task){
+      if (task.title == "CREATE")this.createTemplate(task);
+      else if(task.title == "UPDATE")this.updateTamplate(task);
+      else if(task.title == "DELETE")this.deleteTemplate(task);
+      else if(task.title == "FIND_ONE")this.findOneTemplate(task);
+      else if(task.title == "GET_ALL")this.findAllTemplate(task);
```

Comment:
please consider following issues:
1) prefer strict compare - it is safer
2) there is no handler for unknown command
3) multiple if's can be nicely replaced by strategy pattern